### PR TITLE
Level Bugfix 

### PR DESCRIPTION
**Pull request by AElanagai**
# Description

Closes the issue where the game crashed after player completed level 3 as the game was not set for a winning condition , which is now set to be after level 3 . So when player completes level 3 after reaching the boss room the game shows the win screen rather than loading the level 4 tile asset that caused the game to crash .

# Closes Issue

 #515 

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: source/core/src/main/com/csse3200/game/areas/EnemyRoom.java [hunk 1]
```diff
@@ -194,7 +194,7 @@ protected void spawnAnimals(GameArea area, Entity player, GridPoint2 min, GridPo
                     spawnEnemyEntity(area, enemy, randomPos);
                 }
             }
```
Comment (vsinh14) [inline source/core/src/main/com/csse3200/game/areas/EnemyRoom.java:196]: remove makeallanimalsdead

**Review by vsinh14 — COMMENTED** [review]
Code looks fine without testing

**Comment by vsinh14** [comment]
back to menu screen does not work from win game screen @IanP96

**Comment by IanP96** [comment]
> back to menu screen does not work from win game screen @IanP96

pretty confident this is because of the ESC functionality added that disables stage input so mouse clicks aren't received. Working on it now

**Comment by IanP96** [comment]
> back to menu screen does not work from win game screen @IanP96

Fixed @vsinh14 @AElanagai

**Review by vsinh14 — COMMENTED** [review]
Code changes look good now doing manual testing

**Comment by vsinh14** [comment]
Fix the game over screen so it makes sense (ie it says you win ... then u faint) then show the return to main menu option so it makes sense to the player that they actually did complete the game

**Comment by manya-k** [comment]
1) after defeating the third boss, if u approach the stairs, the stairs disappear and win screen doesn't appear
2) if u kill all animals in the third level and if u go to a new room and come back to boss room, there are more animals in boss room except the boss.
there are bugs in this pr

**Comment by vsinh14** [comment]
> 1. third

Merging the bugfix for saving resolves these issues

**Review by vsinh14 — APPROVED** [review]
All changes look good and work well WE FINALLY CAN WIN THE GAME

**Review by Aryan-S-Gupta — APPROVED** [review]

File: source/core/src/main/com/csse3200/game/services/AlertBoxService.java [hunk 2]
```diff
@@ -75,11 +75,38 @@ protected void result(Object object) {
...
     }
 
+    /**
+     * Displays a game won dialog with an Exit button.
+     *
+     * @param title     Title of the dialog.
+     * @param message   Message to display.
+     * @param listener  Listener to handle Exit response.
+     */
+    public void showGameWonDialog(String title, String message, GameWonListener listener) {
```
Comment (manya-k) [inline source/core/src/main/com/csse3200/game/services/AlertBoxService.java:85]: got being used anywhere

**Review by manya-k — APPROVED** [review]
all bugs are fixed.
some unused methods
but overall good to go

**Pull request merged**